packages/components/src/MessageBar/MessageBar.tsx

```diff
@@ -96,6 +113,27 @@ const MessageBarLayout = forwardRef(
       }
     }, [visibleProp])
 
+    const dismissButton = canDismiss && (
+      <IconButton
+        id={id ? `${id}-iconButton` : undefined}
+        ml="auto"
+        onClick={handleDismiss}
+        icon="Close"
+        size="small"
+        label={`Dismiss ${getIntentLabel(intent)}`}
+        aria-hidden
+      />
+    )
+
+    const actions = primaryButton ? (
+      <>
+        {secondaryButton && secondaryButton}
+        {primaryButton}
+      </>
+    ) : (
```

Comment:
I think the JSX could just look like this, right?

```suggestion
    const actions = primaryButton ? (
      <>
        {secondaryButton}
        {primaryButton}
      </>
    ) : (
```

I could be tripping up, but I think the logic already prevents `secondaryButton` from ever existing without a `primaryButton` since no `primaryButton` means the `dismissButton` is rendered instead.

Reply:
> I think the logic already prevents secondaryButton from ever existing without a primaryButton since no primaryButton means the dismissButton is rendered instead.

Yep that's correct! I did the `secondaryButton && secondaryButton` just because I assumed Typescript would yell at me if I tried to render `secondaryButton` without doing a truthy check for it. But it seems totally cool with that so definitely worth committing your suggestion. Thanks!